c10/core/TensorImpl.h

```diff
@@ -1553,7 +1553,8 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
         "Caffe2 uses a lazy allocation, so you will need to call "
         "mutable_data() or raw_mutable_data() to actually allocate memory.");
     // Caller does the type check.
-    return get_data() + storage_offset_;
+    T* data = get_data();
```

Comment:
@albanD would appreciate an advice: I'm debating between decorating the function to suppress pointer-overflow check vs adding `data ? data + storage_offset_ ? nullptr;` ternary. What do you think is better?

Reply:
Given that we check that the storage is initialized, the data should always be valid here. So this is indeed unnecessary conditioning.
Maybe using a `C10_LIKELY()` would be enough to not hinder perf and be ASAN compliant?

Reply:
Replaced it with `__ubsan_suppress_pointer_overflow__`, but will probably create todo to make creation of empty tensors with non-zero storage offsets harder.